Src/HALALMock/Services/Communication/Ethernet/TCP/ServerSocket.cpp

```diff
@@ -72,209 +47,230 @@ ServerSocket::~ServerSocket(){
 	auto it = std::find(OrderProtocol::sockets.begin(), OrderProtocol::sockets.end(), this);
 	if(it == OrderProtocol::sockets.end()) return;
 	else OrderProtocol::sockets.erase(it);
-	tcp_abort(client_control_block);
-	tcp_abort(server_control_block);
-	while(!tx_packet_buffer.empty()){
-		free(tx_packet_buffer.front());
-		tx_packet_buffer.pop();
-	}
-	while(!rx_packet_buffer.empty()){
-		free(rx_packet_buffer.front());
-		rx_packet_buffer.pop();
+	// Clean all descriptors
+	 if (client_fd != -1) {
+        ::close(client_fd);
+        client_fd = -1;
+    }
+    if (server_socket_fd != -1) {
+        ::close(server_socket_fd);
+        server_socket_fd = -1;
+    }
+	//clean the transmisions buffers
+	while (!tx_packet_buffer.empty()) {
+        tx_packet_buffer.pop();
+    }
+    while (!rx_packet_buffer.empty()) {
+        rx_packet_buffer.pop();
+    }
+	//eliminate the threads
+	if(state == LISTENING){
+		~listening_thread();
+	}else if(state == ACCEPTED){
+		~receive_thread();
```

Comment:
I dont know if closing both in the `close` method and in the destructor will give any problems, because usually doing a double close returns an error

Reply:
also see that in the HALAL the implementation is different for close and for the destructor